### PR TITLE
Add More Device Array API Functions to CUDA Simulator

### PR DESCRIPTION
**Pull request by njwhite**
...moer API coverage means the simulator is more useful for debugging live code.

### CODE REVIEW

File: numba/cuda/tests/cudadrv/test_context_stack.py [hunk 1]
```diff
@@ -21,5 +24,19 @@ def test_gpus_iter(self):
         self.assertGreater(len(gpulist), 0)
 
 
+class TestContextAPI(unittest.TestCase):
```
Comment (sklam) [inline numba/cuda/tests/cudadrv/test_context_stack.py]: Can you add `SerialMixin` to this class?  It is necessary to run correctly when the test runner is given the `-m` multiprocessing flag.

**Comment by sklam** [comment]
The patch looks great.  Just a small change needed for a test class and there are now some merge conflicts.

**Comment by njwhite** [comment]
@sklam fixed!

**Comment by sklam** [comment]
Thanks! merging.

**Pull request merged**